source-azure-blob-storage/main.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+//go:embed schema.json
+var schema string
+
+func handleError(err error) {
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+type config struct {
+	AzureClientID       string         `json:"azureClientID"`
+	AzureClientSecret   string         `json:"azureClientSecret"`
+	AzureTenantID       string         `json:"azureTenantID"`
+	AzureSubscriptionID string         `json:"azureSubscriptionID"`
+	StorageAccountName  string         `json:"storageAccountName"`
+	ContainerName       string         `json:"containerName"`
+	Parser              *parser.Config `json:"parser"`
+}
+
+func (c config) Validate() error {
+	if c.AzureTenantID == "" {
```

Comment:
This can be cleaned up by doing something like this: https://github.com/estuary/connectors/blob/cd7c1b12ed2aa450e24953e2d5df18143f12e4ad/materialize-snowflake/config.go#L145-L155